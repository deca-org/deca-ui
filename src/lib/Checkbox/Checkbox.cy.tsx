import { standardColors } from '../Theme';
import { Test } from '../Utils';
import _cyp from '../../../cypress';
import React from 'react';
import Checkbox from './Checkbox';

describe('components/Checkbox', () => {
  describe('before click', () => {
    it('border-color', () => {
      cy.mount(<Checkbox label="Label" />);

      cy.get('label')
        .before('border-color')
        .should('eq', Test.color('gray600'));
    });
    it('disabled', () => {
      cy.mount(<Checkbox label="Label" disabled />);

      cy.get('label').should('have.css', 'color', Test.color('gray500'));

      cy.get('label')
        .before('border-color')
        .should('eq', Test.color('gray400'));
    });
  });
  describe('colors', () => {
    it('label color', () => {
      cy.mount(<Checkbox label="Label" />);
      cy.get('label').should('have.css', 'color', Test.color('black'));
    });
    standardColors.map((color) => {
      describe(color, () => {
        it('background-color', () => {
          cy.mount(<Checkbox color={color} label="Label" />);

          cy.get('label').click();

          // wait for css transition to finish
          cy.wait(250);
          cy.get('label')
            .before('background-color')
            .should('eq', Test.color(color));
        });
        it('border-color', () => {
          cy.mount(<Checkbox color={color} label="Label" />);

          cy.get('label').click();

          // wait for css transition to finish
          cy.wait(250);
          cy.get('label')
            .before('border-color')
            .should('eq', Test.color(color));
        });
        it('disabled', () => {
          cy.mount(
            <Checkbox color={color} label="Label" initialCheck disabled />
          );

          cy.get('label').before('opacity').should('eq', '0.55');
          cy.get('svg').should('have.css', 'opacity', '0.9');
        });
      });
    });
  });
  describe('sizes', () => {
    it('sm', () => {
      cy.mount(<Checkbox size="sm" label="Label" />);
      cy.get('label').before('width').should('eq', Test.size('2'));
      cy.get('label').before('height').should('eq', Test.size('2'));
      cy.get('label').before('margin-right').should('eq', Test.space('1'));
      cy.get('label').should('have.css', 'font-size', Test.fontSize('caption'));
      cy.get('svg').should('have.css', 'width', Test.size('1'));
    });
    it('md', () => {
      cy.mount(<Checkbox size="md" label="Label" />);
      cy.get('label').before('width').should('eq', Test.size('3'));
      cy.get('label').before('height').should('eq', Test.size('3'));
      cy.get('label').before('margin-right').should('eq', Test.space('2'));
      cy.get('label').should('have.css', 'font-size', Test.fontSize('bodySm'));
      cy.get('svg').should('have.css', 'width', Test.size('2'));
    });
    it('lg', () => {
      cy.mount(<Checkbox size="lg" label="Label" />);
      cy.get('label').before('width').should('eq', Test.size('4'));
      cy.get('label').before('height').should('eq', Test.size('4'));
      cy.get('label').before('margin-right').should('eq', Test.space('2'));
      cy.get('label').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('svg').should('have.css', 'width', Test.size('3'));
    });
  });

  describe('no label', () => {
    describe('should have no margin', () => {
      it('sm', () => {
        cy.mount(<Checkbox size="sm" />);
        cy.get('label').before('margin-right').should('eq', '0px');
      });
      it('md', () => {
        cy.mount(<Checkbox size="md" />);
        cy.get('label').before('margin-right').should('eq', '0px');
      });
      it('sm', () => {
        cy.mount(<Checkbox size="md" />);
        cy.get('label').before('margin-right').should('eq', '0px');
      });
    });
  });

  describe('dark mode', () => {
    it('label color', () => {
      cy.darkMount(<Checkbox label="Label" />);
      cy.get('label').should('have.css', 'color', Test.color('white'));
    });

    it('disabled state', () => {
      cy.darkMount(<Checkbox label="Label" initialCheck disabled />);
      cy.get('label').before('opacity').should('eq', '0.5');
      cy.get('svg').should('have.css', 'opacity', '0.3');
    });
  });
});
