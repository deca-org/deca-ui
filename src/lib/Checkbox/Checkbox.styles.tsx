import { CSS, styled, standardColors } from '@lib/Theme/stitches.config';
import { getStaticColor } from '@lib/Utils';
import { darken } from 'polished';

const compoundVariantComposer = () => {
  const colorVariants: {
    isDisabled?: boolean;
    color?: string;
    isDark?: boolean;
    css: CSS;
  }[] = [];

  standardColors.map((color) => {
    colorVariants.push(
      {
        isDisabled: true,
        isDark: false,
        color: color,
        css: {
          '&:checked:active + label::before': {
            bg: `$${color}`,
            borderColor: `$${color}`,
          },
        },
      },
      {
        isDisabled: true,
        isDark: true,
        color: color,
        css: {
          '&:checked:active + label::before': {
            bg: `$${color}`,
            borderColor: `$${color}`,
          },
        },
      }
    );
  });
  return colorVariants;
};

export const StyledCheckboxWrapper = styled('div', {
  boxSizing: 'border-box',
  display: 'flex',
  alignItems: 'center',
  fontFamily: '$normal',
  transition: '$default',
});

export const StyledCheckbox = styled('input', {
  cursor: 'pointer',
  opacity: 0,
  position: 'absolute',
  transition: '$default',
  '&:checked + label > svg': {
    visibility: 'visible',
    color: '$white',
    opacity: 1,
  },
  compoundVariants: [
    ...compoundVariantComposer(),
    {
      isDisabled: true,
      isDark: true,
      css: {
        '&:checked + label::before': {
          opacity: '50%',
        },
        '&:checked + label > svg': {
          opacity: '30%',
        },
      },
    },
    {
      isDisabled: true,
      isDark: false,
      css: {
        '&:checked + label::before': {
          opacity: '55%',
        },
        '&:checked + label > svg': {
          opacity: '90%',
        },
      },
    },
  ],
  variants: {
    size: {
      sm: {},
      md: {},
      lg: {},
    },
    color: {
      primary: {
        '&:checked + label::before': {
          bg: '$primary',
          borderColor: '$primary',
        },
        '&:checked:active + label::before': {
          bg: '$primary-darken-1',
          borderColor: '$primary-darken-1',
        },
      },
      secondary: {
        '&:checked + label::before': {
          bg: '$secondary',
          borderColor: '$secondary',
        },
        '&:checked:active + label::before': {
          bg: '$secondary-darken-1',
          borderColor: '$secondary-darken-1',
        },
      },
      success: {
        '&:checked + label::before': {
          bg: '$success',
          borderColor: '$success',
        },
        '&:checked:active + label::before': {
          bg: '$success-darken-1',
          borderColor: '$success-darken-1',
        },
      },
      warning: {
        '&:checked + label::before': {
          bg: '$warning',
          borderColor: '$warning',
        },
        '&:checked:active + label::before': {
          bg: '$warning-darken-1',
          borderColor: '$warning-darken-1',
        },
      },
      error: {
        '&:checked + label::before': {
          bg: '$error',
          borderColor: '$error',
        },
        '&:checked:active + label::before': {
          bg: '$error-darken-1',
          borderColor: '$error-darken-1',
        },
      },
    },
    isDisabled: {
      true: {},
      false: {
        '&:focus-visible + label::before': {
          boxShadow: '$a11y',
        },
      },
    },
    isDark: {
      true: {
        '&:checked + label > svg': {
          visibility: 'visible',
          color: '$black',
          opacity: 1,
        },
      },
      false: {},
    },
  },
});

export const StyledCheckboxLabel = styled('label', {
  color: '$text',
  position: 'relative',
  cursor: 'pointer',
  display: 'flex',
  alignItems: 'center',
  transition: '$default',
  '&::before': {
    boxSizing: 'content-box',
    borderColor: '$gray600',
  },
  '& svg': {
    color: '$white',
    opacity: 0,
    transition: '$default',
  },
  compoundVariants: [
    {
      hasLabel: true,
      size: 'sm',
      css: {
        '&::before': {
          mr: '$1',
        },
      },
    },
    {
      hasLabel: true,
      size: 'md',
      css: {
        '&::before': {
          mr: '$2',
        },
      },
    },
    {
      hasLabel: true,
      size: 'lg',
      css: {
        '&::before': {
          mr: '$2',
        },
      },
    },
    {
      isDisabled: true,
      isDark: true,
      css: {
        color: '$gray700',
        '&::before': {
          borderColor: '$gray700',
        },
      },
    },

    {
      isDisabled: true,
      isDark: false,
      css: {
        color: '$gray500',
        '&::before': {
          borderColor: '$gray400',
        },
      },
    },
  ],
  variants: {
    size: {
      sm: {
        fontSize: '$caption',
        '& svg': {
          width: '$1',
          ml: '$1',
          position: 'absolute',
          visibility: 'hidden',
          pt: 'calc($0 / 4)',
        },
        '&::before': {
          transition: '$default',
          content: '',
          size: '$2',
          borderStyle: 'solid',
          borderWidth: '$normal',
          br: '$xs',
          mt: 'calc($0 / 4)',
        },
      },
      md: {
        fontSize: '$bodySm',
        '& svg': {
          width: '$2',
          ml: '$1',
          position: 'absolute',
          visibility: 'hidden',
        },
        '&::before': {
          transition: '$default',
          content: '',
          size: '$3',
          borderStyle: 'solid',
          borderWidth: '$normal',
          br: '$xs',
        },
      },
      lg: {
        fontSize: '$body',
        '& svg': {
          width: '$3',
          ml: '$1',
          position: 'absolute',
          visibility: 'hidden',
        },
        '&::before': {
          transition: '$default',
          content: '',
          size: '$4',
          borderStyle: 'solid',
          borderWidth: '$normal',
          br: '$xs',
        },
      },
    },
    color: {
      primary: {},
      secondary: {},
      success: {},
      warning: {},
      error: {},
    },
    isDisabled: {
      true: {
        cursor: 'not-allowed',
        '-webkit-user-select': 'none',
        '-moz-user-select': 'none',
        '-ms-user-select': 'none',
        userSelect: 'none',
      },
      false: {
        '&:hover::before': {
          transition: '$default',
          borderColor: darken(0.125, getStaticColor('gray600')),
        },
      },
    },
    hasLabel: {
      true: {},
      false: {},
    },
    isDark: {
      true: {},
      false: {},
    },
  },
});
