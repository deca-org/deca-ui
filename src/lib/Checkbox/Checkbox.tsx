import { ThemeContext } from '@lib/Theme';
import { CSS, StandardColors } from '@lib/Theme/stitches.config';
import {
  Modify,
  PolymorphicRef,
  PolymorphicComponentPropWithRef,
  uuid,
  __DEV__,
  MasterComponent,
} from '@lib/Utils';
import clsx from 'clsx';
import React, { useState, useMemo } from 'react';
import CheckboxGroup from './CheckboxGroup';

import {
  StyledCheckboxWrapper,
  StyledCheckbox,
  StyledCheckboxLabel,
} from './Checkbox.styles';

/**
 * Checkboxes can be used to turn an option on or off
 */
interface Props
  extends Modify<
    React.ComponentPropsWithRef<'input'>,
    {
      /**
       * Size of the component.
       * @default md
       */
      size?: 'sm' | 'md' | 'lg';
    }
  > {
  /**
   * Size of the component.
   * @default md
   */
  size?: 'sm' | 'md' | 'lg';
  /**
   * Text label for the component.
   */
  label?: string;
  /**
   * ClassName applied to the component.
   * @default ''
   */
  className?: string;
  /**
   * Disabled state applied to the component.
   * @default false
   */
  disabled?: boolean;
  /**
   * Override default CSS style.
   */
  css?: CSS;
  /**
   * Color of checkbox when active.
   * @default primary
   */
  color?: StandardColors;
  /**
   * The value assigned to the checkbox
   */
  value?: string;
  /**
   * Whether or not the checkbox is checked.
   */
  checked?: boolean;
  /**
   * Function that executes when checkbox is checked or unchecked.
   */
  onChange?(e: React.ChangeEvent<HTMLInputElement>): void;
  /**
   * Callback fired when a checkbox is focused
   */
  onFocus?(e: React.ChangeEvent<HTMLInputElement>): void;
  /**
   * Name is used as an identifier in a form.
   */
  name?: string;
  /**
   *  Whether or not the checkbox is initially checked.
   */
  initialCheck?: boolean;
  /**
   * Required checkbox prop.
   * @default false
   */
  required?: boolean;
}

const Check = () => (
  <svg
    viewBox="0 0 512 512"
    aria-hidden="true"
    focusable="false"
    fill="currentColor"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      fill="currentColor"
      d="m173.898 439.404-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
    ></path>
  </svg>
);

export type CheckboxProps<T extends React.ElementType> =
  PolymorphicComponentPropWithRef<T, Props>;

export type CheckboxComponent = (<C extends React.ElementType = 'input'>(
  props: CheckboxProps<C>
) => React.ReactElement | null) & { displayName?: string };

const Checkbox: CheckboxComponent = React.forwardRef(
  <T extends React.ElementType = 'input'>(
    {
      size = 'md',
      label,
      className,
      name,
      disabled = false,
      as,
      css,
      color = 'primary',
      required = false,
      checked,
      initialCheck = false,
      onChange,
      value,
      ...props
    }: CheckboxProps<T>,
    ref?: PolymorphicRef<T>
  ) => {
    const [selfChecked, setSelfChecked] = useState<boolean>(initialCheck);

    const checkboxId = useMemo(() => {
      if (props.id) {
        return props.id;
      } else {
        return uuid();
      }
    }, [props.id]);

    const isControlledComponent = useMemo(
      () => checked !== undefined,
      [checked]
    );

    const changeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
      if (disabled) return;
      if (!isControlledComponent) {
        setSelfChecked(e.target.checked);
      }
      onChange && onChange(e);
    };

    const preClass = 'decaCheckbox';

    const { dark } = React.useContext(ThemeContext);

    return (
      <StyledCheckboxWrapper className={clsx(className, `${preClass}-root`)}>
        <StyledCheckbox
          id={checkboxId}
          type="checkbox"
          onChange={changeHandler}
          checked={isControlledComponent ? checked : selfChecked}
          ref={ref}
          required={required}
          className={`${preClass}-input`}
          name={name}
          size={size}
          color={color}
          disabled={disabled}
          isDisabled={disabled}
          value={value}
          isDark={dark}
          {...props}
        />
        <StyledCheckboxLabel
          htmlFor={checkboxId}
          css={css}
          size={size}
          as={as as any} // as prop will not accept T in this case
          color={color}
          isDisabled={disabled}
          hasLabel={label ? true : false}
          isDark={dark}
        >
          <Check />
          {label && label}
        </StyledCheckboxLabel>
      </StyledCheckboxWrapper>
    );
  }
);

if (__DEV__) {
  Checkbox.displayName = 'DecaUI.Checkbox';
}

export default Checkbox as MasterComponent<
  HTMLInputElement,
  CheckboxProps<React.ElementType>,
  { Group: typeof CheckboxGroup }
>;
