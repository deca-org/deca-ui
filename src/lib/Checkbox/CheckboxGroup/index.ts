import CheckboxGroup from './CheckboxGroup';
export * from './CheckboxGroup';
export { CheckboxGroup };
export default CheckboxGroup;
