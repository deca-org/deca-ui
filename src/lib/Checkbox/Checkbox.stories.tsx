import { DecaUIProvider } from '@lib/Theme';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import Checkbox from './Checkbox';

export default {
  title: 'Checkbox',
  component: Checkbox,
} as ComponentMeta<typeof Checkbox>;

const Template: ComponentStory<typeof Checkbox> = (args) => (
  <Checkbox {...args} />
);

export const Default = Template.bind({});
Default.args = {
  label: 'Label',
  size: 'md',
  color: 'primary',
  disabled: false,
  css: {},
  className: '',
  initialCheck: true,
  required: false,
};

export const NoLabel = Template.bind({});
NoLabel.args = {
  ...Default.args,
  label: '',
};

export const WithTheme = Template.bind({});

WithTheme.args = { ...Default.args };
WithTheme.decorators = [
  (Story) => (
    <DecaUIProvider
      theme={{
        colors: {
          primary: '$green600',
        },
      }}
    >
      <Story />
    </DecaUIProvider>
  ),
];

export const DarkMode = Template.bind({});

DarkMode.args = { ...Default.args };
DarkMode.parameters = { backgrounds: { default: 'dark' } };

DarkMode.decorators = [
  (Story) => (
    <DecaUIProvider mode="dark">
      <Story />
    </DecaUIProvider>
  ),
];
