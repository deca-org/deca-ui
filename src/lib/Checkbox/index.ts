import Checkbox from './Checkbox';
import CheckboxGroup from './CheckboxGroup';

Checkbox.Group = CheckboxGroup;

export * from './Checkbox';
export { Checkbox };
export default Checkbox;
