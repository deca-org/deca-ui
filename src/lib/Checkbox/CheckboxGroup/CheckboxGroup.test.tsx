import '@testing-library/jest-dom';
import Checkbox from '@lib/Checkbox';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

describe('components/CheckboxGroup', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Checkbox.Group defaultValue={['A', 'B']}>
        <Checkbox value="A" label="Option A" />
        <Checkbox value="B" label="Option B" />
        <Checkbox value="C" label="Option C" />
        <Checkbox value="D" label="Option D" />
      </Checkbox.Group>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('works as an uncontrolled component', async () => {
    const user = userEvent.setup();
    render(
      <Checkbox.Group defaultValue={['A', 'B']}>
        <Checkbox id="A" value="A" label="Option A" />
        <Checkbox id="B" value="B" label="Option B" />
        <Checkbox id="C" value="C" label="Option C" />
        <Checkbox id="D" value="D" label="Option D" />
      </Checkbox.Group>
    );

    expect(screen.getByLabelText('Option A')).toBeChecked();
    expect(screen.getByLabelText('Option B')).toBeChecked();
    await user.click(screen.getByLabelText('Option C'));
    expect(screen.getByLabelText('Option C')).toBeChecked();
  });
  it('works as a controlled component', async () => {
    let value = ['A', 'B'];
    const user = userEvent.setup();

    render(
      <Checkbox.Group
        value={value}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
          (value = [...value, e.target.value])
        }
      >
        <Checkbox id="A" value="A" label="Option A" />
        <Checkbox id="B" value="B" label="Option B" />
        <Checkbox id="C" value="C" label="Option C" />
        <Checkbox id="D" value="D" label="Option D" />
      </Checkbox.Group>
    );

    expect(screen.getByLabelText('Option A')).toBeChecked();
    expect(screen.getByLabelText('Option B')).toBeChecked();
    await user.click(screen.getByLabelText('Option C'));
    expect(value).toStrictEqual(['A', 'B', 'C']);
  });
});
