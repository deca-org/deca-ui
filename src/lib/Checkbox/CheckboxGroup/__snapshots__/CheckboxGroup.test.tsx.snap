// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CheckboxGroup matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="c-HwauS decaCheckboxGroup-root"
  >
    <div
      class="c-gidqah decaCheckbox-root"
    >
      <input
        checked=""
        class="c-eCNcrt c-eCNcrt-fdUhn-color-primary c-eCNcrt-kueLmT-isDisabled-false decaCheckbox-input"
        id="1"
        name="checkbox-1"
        type="checkbox"
        value="A"
      />
      <label
        class="c-flUSLM c-flUSLM-ijjbfZ-size-md c-flUSLM-hqEieG-isDisabled-false c-flUSLM-jQVVMl-cv"
        for="1"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m173.898 439.404-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
        Option A
      </label>
    </div>
    <div
      class="c-gidqah decaCheckbox-root"
    >
      <input
        checked=""
        class="c-eCNcrt c-eCNcrt-fdUhn-color-primary c-eCNcrt-kueLmT-isDisabled-false decaCheckbox-input"
        id="1"
        name="checkbox-1"
        type="checkbox"
        value="B"
      />
      <label
        class="c-flUSLM c-flUSLM-ijjbfZ-size-md c-flUSLM-hqEieG-isDisabled-false c-flUSLM-jQVVMl-cv"
        for="1"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m173.898 439.404-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
        Option B
      </label>
    </div>
    <div
      class="c-gidqah decaCheckbox-root"
    >
      <input
        class="c-eCNcrt c-eCNcrt-fdUhn-color-primary c-eCNcrt-kueLmT-isDisabled-false decaCheckbox-input"
        id="1"
        name="checkbox-1"
        type="checkbox"
        value="C"
      />
      <label
        class="c-flUSLM c-flUSLM-ijjbfZ-size-md c-flUSLM-hqEieG-isDisabled-false c-flUSLM-jQVVMl-cv"
        for="1"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m173.898 439.404-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
        Option C
      </label>
    </div>
    <div
      class="c-gidqah decaCheckbox-root"
    >
      <input
        class="c-eCNcrt c-eCNcrt-fdUhn-color-primary c-eCNcrt-kueLmT-isDisabled-false decaCheckbox-input"
        id="1"
        name="checkbox-1"
        type="checkbox"
        value="D"
      />
      <label
        class="c-flUSLM c-flUSLM-ijjbfZ-size-md c-flUSLM-hqEieG-isDisabled-false c-flUSLM-jQVVMl-cv"
        for="1"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m173.898 439.404-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
        Option D
      </label>
    </div>
  </div>
</DocumentFragment>
`;
